spyder/utils/environ.py

```diff
@@ -97,90 +144,56 @@ def __init__(self, environ, parent=None):
 
 class EnvDialog(RemoteEnvDialog):
     """Environment variables Dialog"""
+
     def __init__(self, parent=None):
         RemoteEnvDialog.__init__(self, dict(os.environ), parent=parent)
 
 
-# For Windows only
-try:
-    from spyder.py3compat import winreg
-
-    def get_user_env():
-        """Return HKCU (current user) environment variables"""
-        reg = dict()
-        key = winreg.OpenKey(winreg.HKEY_CURRENT_USER, "Environment")
-        for index in range(0, winreg.QueryInfoKey(key)[1]):
-            try:
-                value = winreg.EnumValue(key, index)
-                reg[value[0]] = value[1]
-            except:
-                break
-        return envdict2listdict(reg)
-
-    def set_user_env(reg, parent=None):
-        """Set HKCU (current user) environment variables"""
-        reg = listdict2envdict(reg)
-        types = dict()
-        key = winreg.OpenKey(winreg.HKEY_CURRENT_USER, "Environment")
-        for name in reg:
-            try:
-                _x, types[name] = winreg.QueryValueEx(key, name)
-            except WindowsError:
-                types[name] = winreg.REG_EXPAND_SZ
-        key = winreg.OpenKey(winreg.HKEY_CURRENT_USER, "Environment", 0,
-                             winreg.KEY_SET_VALUE)
-        for name in reg:
-            winreg.SetValueEx(key, name, 0, types[name], reg[name])
-        try:
-            from win32gui import SendMessageTimeout
-            from win32con import (HWND_BROADCAST, WM_SETTINGCHANGE,
-                                  SMTO_ABORTIFHUNG)
-            SendMessageTimeout(HWND_BROADCAST, WM_SETTINGCHANGE, 0,
-                               "Environment", SMTO_ABORTIFHUNG, 5000)
-        except Exception:
-            QMessageBox.warning(parent, _("Warning"),
-                        _("Module <b>pywin32 was not found</b>.<br>"
-                          "Please restart this Windows <i>session</i> "
-                          "(not the computer) for changes to take effect."))
-
-    class WinUserEnvDialog(CollectionsEditor):
-        """Windows User Environment Variables Editor"""
-        def __init__(self, parent=None):
-            super(WinUserEnvDialog, self).__init__(parent)
-            self.setup(get_user_env(),
-                       title=r"HKEY_CURRENT_USER\Environment")
+class UserEnvDialog(RemoteEnvDialog):
+    """User Environment Variables Viewer/Editor"""
+
+    def __init__(self, parent=None):
+        title = _("User Environment variables")
+        readonly = True
+        if os.name == 'nt':
+            title = _(r"HKEY_CURRENT_USER\Environment")
```

Comment:
```suggestion
            title = _(r"User environment variables in Windows registry")
```

I doubt many users understand what `HKEY_CURRENT_USER` is. However, I think my suggestion better says what we want to convey here.